src/server/object_services/object_server.js

```diff
@@ -114,6 +115,8 @@ async function create_object_upload(req) {
     }
 
     const tier = req.bucket.tiering && await map_server.select_tier_for_write(req.bucket);
+    if (tier.storage_class !== STORAGE_CLASS_STANDARD) info.storage_class = tier.storage_class;
```

Comment:
DeepScan fix before this I guess. Need to do a null check. Or maybye
```suggestion
    if (tier?.storage_class !== STORAGE_CLASS_STANDARD) info.storage_class = tier?.storage_class;
```